external-crates/move/crates/move-compiler/src/cfgir/visitor.rs

```diff
@@ -41,7 +55,247 @@ pub trait AbstractInterpreterVisitor {
+    /// custom version should of this function should be created if different type of analysis is
+    /// required.
+    fn visit(&mut self, program: &mut G::Program) {
+        for (mident, mdef) in program.modules.key_cloned_iter_mut() {
+            self.add_warning_filter_scope(mdef.warning_filter.clone());
+            if self.visit_module_custom(mident, mdef) {
+                self.pop_warning_filter_scope();
+                continue;
+            }
+
+            for (struct_name, sdef) in mdef.structs.key_cloned_iter_mut() {
+                self.visit_struct(mident, struct_name, sdef)
+            }
+            for (enum_name, edef) in mdef.enums.key_cloned_iter_mut() {
+                self.visit_enum(mident, enum_name, edef)
+            }
+            for (constant_name, cdef) in mdef.constants.key_cloned_iter_mut() {
+                self.visit_constant(mident, constant_name, cdef)
+            }
+            for (function_name, fdef) in mdef.functions.key_cloned_iter_mut() {
+                self.visit_function(mident, function_name, fdef)
+            }
+
+            self.pop_warning_filter_scope();
+        }
+    }
+
+    // TODO  type visiting
+
+    fn visit_struct_custom(
+        &mut self,
+        _module: ModuleIdent,
+        _struct_name: DatatypeName,
+        _sdef: &mut H::StructDefinition,
+    ) -> bool {
+        false
+    }
+    fn visit_struct(
+        &mut self,
+        module: ModuleIdent,
+        struct_name: DatatypeName,
+        sdef: &mut H::StructDefinition,
+    ) {
+        self.add_warning_filter_scope(sdef.warning_filter.clone());
+        if self.visit_struct_custom(module, struct_name, sdef) {
+            self.pop_warning_filter_scope();
+            return;
+        }
+        self.pop_warning_filter_scope();
+    }
+
+    fn visit_enum_custom(
+        &mut self,
+        _module: ModuleIdent,
+        _enum_name: DatatypeName,
+        _edef: &mut H::EnumDefinition,
+    ) -> bool {
+        false
+    }
+    fn visit_enum(
+        &mut self,
+        module: ModuleIdent,
+        enum_name: DatatypeName,
+        edef: &mut H::EnumDefinition,
+    ) {
+        self.add_warning_filter_scope(edef.warning_filter.clone());
+        if self.visit_enum_custom(module, enum_name, edef) {
+            self.pop_warning_filter_scope();
+            return;
+        }
+        self.pop_warning_filter_scope();
+    }
+
+    fn visit_constant_custom(
+        &mut self,
+        _module: ModuleIdent,
+        _constant_name: ConstantName,
+        _cdef: &mut G::Constant,
+    ) -> bool {
+        false
+    }
+    fn visit_constant(
+        &mut self,
+        module: ModuleIdent,
+        constant_name: ConstantName,
+        cdef: &mut G::Constant,
+    ) {
+        self.add_warning_filter_scope(cdef.warning_filter.clone());
+        if self.visit_constant_custom(module, constant_name, cdef) {
+            self.pop_warning_filter_scope();
+            return;
+        }
```

Comment:
Do we want to add `visit_value` and call that here?

Reply:
Sure, seems helpful in the CFGIR